nca/Parsers/IstioGatewayPolicyGenerator.py

```diff
@@ -0,0 +1,293 @@
+        :param is_ingress: whether this is an ingress (True) or egress (False) rule
+        :return: the resulting GatewayPolicyRule
+        """
+        conn_cube = this_route_conn_cube.copy()
+        conn_cube["dst_ports"] = dest.ports
+        conns = ConnectionSet()
+        conns.add_connections(self.protocol_name, ConnectivityProperties.make_conn_props(conn_cube))
+        conn_cube.update({"src_peers": source_peers, "dst_peers": dest.pods, "protocols": self.protocols})
+        opt_props = ConnectivityProperties.make_conn_props(conn_cube)
+        if is_ingress:
+            return GatewayPolicyRule(source_peers, conns, opt_props)
+        else:
+            return GatewayPolicyRule(dest.pods, conns, opt_props)
+
+    @staticmethod
+    def create_deny_rule(source_peers, dst_peers):
+        """
+        Create a rule for representation of the denied connections from between given mesh and DNS nodes
+        :return: the resulting GatewayPolicyRule
+        """
+        opt_props = ConnectivityProperties.make_conn_props_from_dict({"src_peers": source_peers,
+                                                                      "dst_peers": dst_peers})
+        return GatewayPolicyRule(dst_peers, ConnectionSet(True), opt_props)
+
+    def create_gtw_to_mesh_and_deny_policies(self, vs, route, route_cnt, gtw_to_hosts, used_gateways):
+        """
+        Create internal policies representing connections from gateways to mesh,
+            described by the given http/tls/tcp route.
+        When relevant, create also deny policies from mesh to the mentioned DNS entries.
+        :param VirtualService vs: the virtual service holding the given route
+        :param Route route: the given http/tls/tcp route from which policies should be created
+        :param int route_cnt: the index (starting from 1) if the given route inside this virtual service routes
+        :param dict(Gateway, MinDFA) gtw_to_hosts: a map from gateways to hosts relevant for this route
+        :param set(Gateway) used_gateways: a set of used gateways, to be updated by adding gateways
+               referenced by these routes
+        :return: list[GatewayPolicy] the resulting list of policies
+        """
+        result = []
+        this_route_conn_cube = self.init_route_conn_cube(route)
+        for gtw, host_dfa in gtw_to_hosts.items():
+            # Modeling connections from ingress gateway nodes to internal service nodes (Ingress flow) or
+            # from egress gateway nodes to external service nodes (DNS nodes) (Egress flow).
+            if route.all_sni_hosts_dfa:
+                host_dfa &= route.all_sni_hosts_dfa
+                if not host_dfa:
+                    continue
+            this_route_conn_cube["hosts"] = host_dfa
+            used_gateways.add(gtw)
+            allow_policy = self.init_gtw_to_mesh_policy(vs, route_cnt, gtw)
+            deny_policy = None
+            if gtw.type == Gateway.GatewayType.Egress:
```

Comment:
why don't you check `GatewayType.Ingress` for the ingress gateway policies?

Reply:
This `if` is only relevant for deny policies, which are created only for the egress flow. I moved the comment to the condition line above, hopefully it will be more clear.